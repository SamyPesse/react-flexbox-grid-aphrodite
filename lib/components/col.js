'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

exports.default = Col;

var _react = require('react');

var _aphroditeToJss = require('aphrodite-to-jss');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _createProps = require('./create-props');

var _createProps2 = _interopRequireDefault(_createProps);

var _style = require('./style');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var ModificatorType = _propTypes2.default.oneOfType([_propTypes2.default.number, _propTypes2.default.bool]);

/* eslint-disable react/no-unused-prop-types */
var propTypes = {
  xs: ModificatorType,
  sm: ModificatorType,
  md: ModificatorType,
  lg: ModificatorType,
  xsOffset: _propTypes2.default.number,
  smOffset: _propTypes2.default.number,
  mdOffset: _propTypes2.default.number,
  lgOffset: _propTypes2.default.number,
  xsFirst: _propTypes2.default.bool,
  smFirst: _propTypes2.default.bool,
  mdFirst: _propTypes2.default.bool,
  lgFirst: _propTypes2.default.bool,
  xsLast: _propTypes2.default.bool,
  smLast: _propTypes2.default.bool,
  mdLast: _propTypes2.default.bool,
  lgLast: _propTypes2.default.bool,
  reverse: _propTypes2.default.bool,
  className: _propTypes2.default.string,
  // aphrodite-to-jss SheetDefinitions
  styles: _propTypes2.default.any,
  tagName: _propTypes2.default.string,
  children: _propTypes2.default.node
};
/* eslint-enable */

var classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
  xsFirst: 'first-xs',
  smFirst: 'first-sm',
  mdFirst: 'first-md',
  lgFirst: 'first-lg',
  xsLast: 'last-xs',
  smLast: 'last-sm',
  mdLast: 'last-md',
  lgLast: 'last-lg'
};

function getClassNames(props) {
  var extraClasses = [];

  if (props.reverse) {
    extraClasses.push(_style2.default.reverse);
  }

  var classes = Object.keys(props).filter(function (key) {
    return classMap[key];
  }).map(function (key) {
    return _style2.default[Number.isInteger(props[key]) ? classMap[key] + '-' + props[key] : classMap[key]];
  }).concat(extraClasses).filter(function (k) {
    return (typeof k === 'undefined' ? 'undefined' : _typeof(k)) === 'object';
  }).concat([props.styles]);

  return !props.className ? _aphroditeToJss.css.apply(undefined, _toConsumableArray(classes)) : props.className + ' ' + _aphroditeToJss.css.apply(undefined, _toConsumableArray(classes));
}

function Col(props) {
  var className = getClassNames(props);

  return (0, _react.createElement)(props.tagName || 'div', (0, _createProps2.default)(propTypes, props, className));
}

Col.propTypes = propTypes;
module.exports = exports['default'];