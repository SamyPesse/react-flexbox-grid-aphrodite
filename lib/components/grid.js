'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = Grid;

var _react = require('react');

var _aphroditeToJss = require('aphrodite-to-jss');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _createProps = require('./create-props');

var _createProps2 = _interopRequireDefault(_createProps);

var _style = require('../components/style');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-disable react/no-unused-prop-types */
var propTypes = {
  fluid: _propTypes2.default.bool,
  className: _propTypes2.default.string,
  tagName: _propTypes2.default.string,
  // aphrodite-to-jss SheetDefinitions
  styles: _propTypes2.default.any,
  children: _propTypes2.default.node
};
/* eslint-enable */

function Grid(props) {
  var containerClass = (0, _aphroditeToJss.css)(_style2.default[props.fluid ? 'container-fluid' : 'container'], props.styles);
  var className = [props.className, containerClass].filter(function (o) {
    return o;
  }).join(' ');

  return (0, _react.createElement)(props.tagName || 'div', (0, _createProps2.default)(propTypes, props, className));
}

Grid.propTypes = propTypes;
module.exports = exports['default'];