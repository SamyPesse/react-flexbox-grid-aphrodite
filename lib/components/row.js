'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _typeof = typeof Symbol === "function" && typeof Symbol.iterator === "symbol" ? function (obj) { return typeof obj; } : function (obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; };

exports.default = Row;

var _react = require('react');

var _aphroditeToJss = require('aphrodite-to-jss');

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _createProps = require('./create-props');

var _createProps2 = _interopRequireDefault(_createProps);

var _style = require('./style');

var _style2 = _interopRequireDefault(_style);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }

var modificatorKeys = ['start', 'center', 'end', 'top', 'middle', 'bottom', 'around', 'between', 'first', 'last'];

/* eslint-disable react/no-unused-prop-types */
var propTypes = {
  reverse: _propTypes2.default.bool,
  start: _propTypes2.default.string,
  center: _propTypes2.default.string,
  end: _propTypes2.default.string,
  top: _propTypes2.default.string,
  middle: _propTypes2.default.string,
  bottom: _propTypes2.default.string,
  around: _propTypes2.default.string,
  between: _propTypes2.default.string,
  first: _propTypes2.default.string,
  last: _propTypes2.default.string,
  className: _propTypes2.default.string,
  // aphrodite-to-jss SheetDefinitions
  styles: _propTypes2.default.any,
  tagName: _propTypes2.default.string,
  children: _propTypes2.default.node
};
/* eslint-enable */

function getClassNames(props) {
  var modificators = [_style2.default.row];

  // eslint-disable-next-line no-plusplus

  var _loop = function _loop(i) {
    var key = modificatorKeys[i];
    var value = props[key];
    if (value) {
      value.split(',').forEach(function (m) {
        modificators.push(_style2.default[key + '-' + m]);
      });
    }
  };

  for (var i = 0; i < modificatorKeys.length; ++i) {
    _loop(i);
  }

  if (props.reverse) {
    modificators.push(_style2.default.reverse);
  }

  var classes = modificators.filter(function (o) {
    return o && (typeof o === 'undefined' ? 'undefined' : _typeof(o)) === 'object';
  }).concat([props.styles]);

  return !props.className ? _aphroditeToJss.css.apply(undefined, _toConsumableArray(classes)) : props.className + ' ' + _aphroditeToJss.css.apply(undefined, _toConsumableArray(classes));
}

function Row(props) {
  return (0, _react.createElement)(props.tagName || 'div', (0, _createProps2.default)(propTypes, props, getClassNames(props)));
}

Row.propTypes = propTypes;
module.exports = exports['default'];