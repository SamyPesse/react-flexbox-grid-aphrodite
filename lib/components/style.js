'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _aphroditeToJss = require('aphrodite-to-jss');

var _grid = require('../grid/grid');

var _grid2 = _interopRequireDefault(_grid);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Create the styles and store them in memory so we
// can reference them in the future
exports.default = _aphroditeToJss.StyleSheet.create((0, _grid2.default)());
module.exports = exports['default'];