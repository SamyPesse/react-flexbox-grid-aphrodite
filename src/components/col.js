import { createElement } from 'react';
import { css } from 'aphrodite-to-jss';
import PropTypes from 'prop-types';
import createProps from './create-props';
import style from './style';

const ModificatorType = PropTypes.oneOfType([PropTypes.number, PropTypes.bool]);

/* eslint-disable react/no-unused-prop-types */
const propTypes = {
  xs: ModificatorType,
  sm: ModificatorType,
  md: ModificatorType,
  lg: ModificatorType,
  xsOffset: PropTypes.number,
  smOffset: PropTypes.number,
  mdOffset: PropTypes.number,
  lgOffset: PropTypes.number,
  xsFirst: PropTypes.bool,
  smFirst: PropTypes.bool,
  mdFirst: PropTypes.bool,
  lgFirst: PropTypes.bool,
  xsLast: PropTypes.bool,
  smLast: PropTypes.bool,
  mdLast: PropTypes.bool,
  lgLast: PropTypes.bool,
  reverse: PropTypes.bool,
  className: PropTypes.string,
  // aphrodite-to-jss SheetDefinitions
  styles: PropTypes.any,
  tagName: PropTypes.string,
  children: PropTypes.node,
};
/* eslint-enable */

const classMap = {
  xs: 'col-xs',
  sm: 'col-sm',
  md: 'col-md',
  lg: 'col-lg',
  xsOffset: 'col-xs-offset',
  smOffset: 'col-sm-offset',
  mdOffset: 'col-md-offset',
  lgOffset: 'col-lg-offset',
  xsFirst: 'first-xs',
  smFirst: 'first-sm',
  mdFirst: 'first-md',
  lgFirst: 'first-lg',
  xsLast: 'last-xs',
  smLast: 'last-sm',
  mdLast: 'last-md',
  lgLast: 'last-lg',
};

function getClassNames(props) {
  const extraClasses = [];

  if (props.reverse) {
    extraClasses.push(style.reverse);
  }

  const classes = Object.keys(props)
    .filter(key => classMap[key])
    .map(key => style[Number.isInteger(props[key]) ? (`${classMap[key]}-${props[key]}`) : classMap[key]])
    .concat(extraClasses)
    .filter(k => typeof k === 'object')
    .concat([props.styles]);

  return !props.className ? css(...classes) : `${props.className} ${css(...classes)}`;
}

export default function Col(props) {
  const className = getClassNames(props);

  return createElement(props.tagName || 'div', createProps(propTypes, props, className));
}

Col.propTypes = propTypes;
