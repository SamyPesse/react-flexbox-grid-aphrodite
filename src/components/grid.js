import { createElement } from 'react';
import { css } from 'aphrodite/no-important';
import PropTypes from 'prop-types';
import createProps from './create-props';
import style from '../components/style';

/* eslint-disable react/no-unused-prop-types */
const propTypes = {
  fluid: PropTypes.bool,
  className: PropTypes.string,
  tagName: PropTypes.string,
  // Aphrodite SheetDefinitions
  styles: PropTypes.any,
  children: PropTypes.node,
};
/* eslint-enable */

export default function Grid(props) {
  const containerClass = css(style[
    props.fluid ? 'container-fluid' : 'container'],
    props.styles,
  );
  const className = [props.className, containerClass].filter(o => o).join(' ');

  return createElement(props.tagName || 'div', createProps(propTypes, props, className));
}

Grid.propTypes = propTypes;
