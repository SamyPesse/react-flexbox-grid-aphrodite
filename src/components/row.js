import { createElement } from 'react';
import { css } from 'aphrodite/no-important';
import PropTypes from 'prop-types';
import createProps from './create-props';
import style from './style';

const modificatorKeys = ['start', 'center', 'end', 'top', 'middle', 'bottom', 'around', 'between', 'first', 'last'];

/* eslint-disable react/no-unused-prop-types */
const propTypes = {
  reverse: PropTypes.bool,
  start: PropTypes.string,
  center: PropTypes.string,
  end: PropTypes.string,
  top: PropTypes.string,
  middle: PropTypes.string,
  bottom: PropTypes.string,
  around: PropTypes.string,
  between: PropTypes.string,
  first: PropTypes.string,
  last: PropTypes.string,
  className: PropTypes.string,
  // Aphrodite SheetDefinitions
  styles: PropTypes.any,
  tagName: PropTypes.string,
  children: PropTypes.node,
};
/* eslint-enable */

function getClassNames(props) {
  const modificators = [style.row];

  // eslint-disable-next-line no-plusplus
  for (let i = 0; i < modificatorKeys.length; ++i) {
    const key = modificatorKeys[i];
    const value = props[key];
    if (value) {
      value.split(',').forEach((m) => {
        modificators.push(style[`${key}-${m}`]);
      });
    }
  }

  if (props.reverse) {
    modificators.push(style.reverse);
  }

  const classes = modificators.filter(o => o && typeof o === 'object').concat([props.styles]);

  return !props.className ? css(...classes) : `${props.className} ${css(...classes)}`;
}

export default function Row(props) {
  return createElement(props.tagName || 'div', createProps(propTypes, props, getClassNames(props)));
}

Row.propTypes = propTypes;
