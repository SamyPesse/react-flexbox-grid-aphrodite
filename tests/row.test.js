import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { css } from 'aphrodite';

import { Row, style } from '../src/components';

describe('Row', () => {
  it('Should add "row" class', () => {
    const wrapper = shallow(<Row />);
    expect(wrapper.props().className).toEqual(css(style.row));
  });

  it('Should add "reverse" class if "reverse" property is true', () => {
    const wrapper = shallow(<Row reverse />);
    expect(wrapper.props().className).toContain(css(style.reverse));
  });

  it('Should preserve class', () => {
    const wrapper = shallow(<Row className="foo" />);
    const { className } = wrapper.props();
    expect(className).toContain('foo');
    expect(className).toContain(css(style.row));
  });

  it('Should add modificators', () => {
    const tree = renderer.create(
      <Row
        start="xs"
        center="sm"
        end="md"
        top="lg"
        middle="xs"
        bottom="sm"
        around="md"
        between="lg"
        first="xs"
        last="sm"
      />,
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('Should support custom tag name', () => {
    const wrapper = shallow(<Row tagName="ul" />);
    expect(wrapper.type()).toBe('ul');
  });
});
