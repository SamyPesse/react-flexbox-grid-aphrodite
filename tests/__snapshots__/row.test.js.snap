exports[`Row Should add modificators 1`] = `
<div
  className="row_1gqinwe-o_O-start-xs_1vpirt5-o_O-start-sm_11eqh8n-o_O-start-md_a0xzfu-o_O-start-lg_1hawvry-o_O-center-xs_12y3xoe-o_O-center-sm_152rggo-o_O-center-md_1v557ds-o_O-center-lg_kcldts-o_O-end-xs_kx5cr3-o_O-end-sm_m6gj9u-o_O-end-md_4r64ii-o_O-end-lg_z3tua0-o_O-top-xs_jd8lz2-o_O-top-sm_i08js-o_O-top-md_10nwr2r-o_O-top-lg_8k763r-o_O-middle-xs_19zvu2s-o_O-middle-sm_gk67xm-o_O-middle-md_1pdxqm8-o_O-middle-lg_15c15ck-o_O-bottom-xs_1yj652h-o_O-bottom-sm_12olb77-o_O-bottom-md_vplhxy-o_O-bottom-lg_192kqq9-o_O-around-xs_1y4agpg-o_O-around-sm_1k6r59i-o_O-around-md_p9gi4p-o_O-around-lg_a70m97-o_O-between-xs_1178gq1-o_O-between-sm_muwqr3-o_O-between-md_17lhkfy-o_O-between-lg_z13jfy-o_O-first-xs_1y522tc-o_O-first-sm_l70vjv-o_O-first-md_sl9qzq-o_O-first-lg_qf0cu9-o_O-last-xs_1au6crx-o_O-last-sm_ezqnk6-o_O-last-md_197773n-o_O-last-lg_upx7rd" />
`;
